libdrgn/drgn.h.in

```diff
@@ -2754,4 +2754,87 @@ struct drgn_error *drgn_object_stack_trace(const struct drgn_object *obj,
 
 /** @} */
 
+/**
+ * @defgroup Threads Threads
+ *
+ * Individual threads resident within a process.
+ *
+ * @{
+ */
+
+/**
+ * @struct drgn_thread
+ *
+ * An individual thread resident within a process
+ */
+struct drgn_thread;
+
+/**
+ * Create a copy of a thread. The thread should be destroyed with @ref drgn_thread_destroy.
+ *
+ * @param[in] thread The thread to be copied.
+ * @param[out] ret The newly allocated copy.
+ * @return @c NULL on success, non-@c NULL on error.
+ */
+struct drgn_error *drgn_thread_dup(const struct drgn_thread *thread,
+				   struct drgn_thread **ret);
+
+void drgn_thread_destroy(struct drgn_thread *thread);
+
+/**
+ * @struct drgn_thread_iterator
+ *
+ * An iterator over all the threads in a target program.
+ */
+struct drgn_thread_iterator;
+
+/**
+ * Find the thread with the given thread ID.
+ *
+ * @param[out] ret The corresponding thread.
```

Comment:
This needs to be freed with `drgn_thread_destroy()`, right? We should document that.